File: algorithm/BaekJoon/BaekJoon_11404.java
package BaekJoon;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BaekJoon_11404{
    static int INF = 10000000;
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int n = Integer.parseInt(br.readLine())+1;

        int[][] arr = new int[n][n];
        int size = Integer.parseInt(br.readLine());

        for(int i = 1; i < n; i++){
            for(int j = 1; j < n; j++){
                if(i!=j && arr[i][j] == 0){
                    arr[i][j] = INF;
                }
            }
        }


        for(int i = 0; i < size; i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            while(st.hasMoreTokens()){
                int val1 = Integer.parseInt(st.nextToken());
                int val2 = Integer.parseInt(st.nextToken());
                int value = Integer.parseInt(st.nextToken());

                if(arr[val1][val2] != 0){
                    arr[val1][val2] = Math.min(arr[val1][val2], value);
                }
                else{
                    arr[val1][val2] = value;
                }
            }
        }

        floyd(arr);

        for(int i = 1; i < arr.length; i++){
            for(int j = 1; j < arr.length; j++){
                System.out.print(arr[i][j]+" ");
            }
            System.out.println();
        }
    }

    private static void floyd(int[][]arr){
        for(int k = 1; k < arr.length; k++){
            for(int i = 1; i < arr.length; i++){
                for(int j = 1; j < arr.length; j++){
                    arr[i][j] = Math.min(arr[i][j], arr[i][k]+arr[k][j]);
                }
            }
        }
        for(int i = 1; i < arr.length; i++){
            for(int j = 1; j < arr.length; j++){
                if(arr[i][j] == INF){
                    arr[i][j] = 0;
                }
            }
        }
    }
}

File: algorithm/BaekJoon/Kotlin/BaekJoon_11404.kt
import java.util.*
var INF = 10000000
fun main() = with(System.`in`.bufferedReader()){
    var n = readLine().toInt() + 1

    var arr = Array(n) { Array(n) { 0 } }

    var size = readLine().toInt()

    for(i in 1 until n){
        for(j in 1 until n){
            if(i!=j && arr[i][j] ==0){
                arr[i][j] = INF
            }
        }
    }

    for(i in 0 until size){
        var st = StringTokenizer(readLine())
        while(st.hasMoreTokens()){
            var val1 = st.nextToken().toInt()
            var val2 = st.nextToken().toInt()
            var value = st.nextToken().toInt()

            if(arr[val1][val2] != 0){
                arr[val1][val2] = Math.min(arr[val1][val2], value)
            }
            else arr[val1][val2] = value
        }
    }
    floyd(arr)

    for(i in 1 until arr.size){
        for(j in 1 until arr.size){
            print("${arr[i][j]} ")
        }
        println()
    }
}

fun floyd(arr: Array<Array<Int>>){
    for(k in 1 until arr.size){
        for(i in 1 until arr.size){
            for(j in 1 until arr.size){
                arr[i][j] = Math.min(arr[i][j], arr[i][k]+arr[k][j])
            }
        }
    }

    for(i in 1 until arr.size){
        for(j in 1 until arr.size){
            if(arr[i][j] == INF){
                arr[i][j] = 0
            }
        }
    }
}